lib/board.rb

```diff
@@ -2,21 +2,84 @@ class Board
 
   def initialize(side_length)
     @side_length = side_length
-    @board = Array.new(@side_length * @side_length, "")
+    @all_moves = set_all_moves(@side_length)
+    @player_moves = Hash.new
+  end
+
+  def to_string_array
+    string_array = Array.new(@all_moves.size, "")
+
+    @player_moves.each do |token, moves_set|
+      assign_moves_to_array(string_array, token, moves_set)
+    end
+
+    string_array
   end
 
   def move(move, player_token)
-    move_index = move - 1
+    if !@player_moves.has_key?(player_token)
+      @player_moves[player_token] = Set.new [move] 
+    else
+      @player_moves[player_token].add(move)
+    end
+  end
```

Comment:
This is an interesting way of building out the board class. This puts a requirement on the board that the board has to have "players" in order for it to exist. Now, if we wanted to reuse your board class for a 1 player game... like sudoku, they would have to worry about having a "player_token". How can you make it so that this board class is more reusable?

Reply:
I would have to figure out how to deal with the number of players in a game (maybe pass in that value in the constructor). I would also need to generalize the "moves" that a player can make on a board (being able to enter integer moves onto the board rather than strings, for Sudoku for example).